### ISSUE TITLE
Deprecate twisted.enterprise.row

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2387]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#2387|
|Type|enhancement|
|Created|2007-01-11 14:18:53Z|

twisted.enterprise.row is unfinished and unmaintained, and should probably be removed.

<details><summary>Searchable metadata</summary>

```
trac-id__2387 2387
type__enhancement enhancement
reporter__itamarst itamarst
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1168525133000000 1168525133000000
changetime__1184570900000000 1184570900000000
version__None None
owner__ 
cc__therve cc__itamarst
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve removed owner|
|-|-|

Ready to review in row-deprecation-2387.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @therve|
|-|-|

1. The row howto should note the deprecation.
2. test_adbapi changes seem irrelevant (if you're fixing logging deprecation warnings, maybe have a ticket for closing those across the board?)
3. Possibly reflector should be deprecated along the way? Is it used for anything else?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

Replying to [itamarst](#note_3):
> 1. The row howto should note the deprecation.

That looks reasonable.

> 2. test_adbapi changes seem irrelevant (if you're fixing logging deprecation warnings, maybe have a ticket for closing those across the board?)

It's related, it's a fix in makeSQLTests, which is used in test_reflector.

> 3. Possibly reflector should be deprecated along the way? Is it used for anything else?

I don't think so.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve removed owner|
|-|-|

I had the deprecation in reflector, and added a small note in the documentation.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @therve|
|-|-|

The row.xhtml note should probably follow the example set by doc/web/howto/woven.xhtml for its markup.  The rest of this branch looks good to me.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve removed owner|
|-|-|

Thanks, I didn't found an example in the core documentation previously.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @therve|
|-|-|

Looks good

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve set status to `closed`|
|-|-|

(In [20746]) Merge row-deprecation-2387

Author: therve
Reviewers: itamarst, exarkun
Fixes [#2387](https://github.com/twisted/twisted/issues/2387)

Deprecate twisted.enterprise.row and twisted.enterprise.reflector.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
